src/main/scala/io/delta/execution/DeltaTableOperations.scala

```diff
@@ -68,4 +74,159 @@ trait DeltaTableOperations { self: DeltaTable =>
+   * Java friendly API.
+   * @param set
+   */
+  def updateExpr(set: java.util.Map[String, String]): Unit = {
+    val setColumns = getSetColumnsForJava(set)
+    executeUpdate(setColumns, None)
+  }
+
+  /**
+   * Update data from the table on the rows that match the given `condition`,
+   * which performs the rules defined by `set`.
+   * @param condition
+   * @param set
+   */
+  def updateExpr(condition: String, set: Map[String, String]): Unit = {
+    val setColumns = getSetColumns(set)
+    executeUpdate(setColumns, Some(functions.expr(condition)))
+  }
+
+  /**
+   * Update data from the table on the rows that match the given `condition`,
+   * which performs the rules defined by `set`.
+   * Java friendly API.
+   * @param condition
+   * @param set
+   */
+  def updateExpr(condition: String, set: java.util.Map[String, String]): Unit = {
+    val setColumns = getSetColumnsForJava(set)
+    executeUpdate(setColumns, Some(functions.expr(condition)))
+  }
+
+  protected def getSetColumns(set: Map[String, String]): Seq[(String, Column)] = {
+    var setColumns: Seq[(String, Column)] = Seq()
+    for ((col, expr) <- set) {
+      setColumns = setColumns :+ ((col, functions.expr(expr)))
+    }
+    setColumns
+  }
+
+  protected def getSetColumnsForColumn(set: Map[String, Column]): Seq[(String, Column)] = {
+    var setColumns: Seq[(String, Column)] = Seq()
+    for ((col, expr) <- set) {
+      setColumns = setColumns :+ ((col, expr))
+    }
+    setColumns
+  }
+
+  protected def getSetColumnsForJava(set: java.util.Map[String, String]): Seq[(String, Column)] = {
+    import scala.collection.JavaConverters._
+    var setColumns: Seq[(String, Column)] = Seq()
+    for ((col, expr) <- set.asScala) {
+      setColumns = setColumns :+ ((col, functions.expr(expr)))
+    }
+    setColumns
+  }
+
+  protected def getSetColumnsForJavaColumn(
+      set: java.util.Map[String, Column]): Seq[(String, Column)] = {
+    import scala.collection.JavaConverters._
+    var setColumns: Seq[(String, Column)] = Seq()
+    for ((col, expr) <- set.asScala) {
+      setColumns = setColumns :+ ((col, expr))
+    }
+    setColumns
+  }
+
+  protected def makeUpdateTable(
+      sparkSession: SparkSession,
+      target: DeltaTable,
+      onCondition: Option[Column],
+      setColumns: Seq[(String, Column)]): UpdateTable = {
+    val updateColumns = setColumns.map { x => UnresolvedAttribute.quotedString(x._1) }
+    val updateExpressions = setColumns.map{ x => x._2.expr }
+    onCondition match {
+      case Some(c) =>
+        new UpdateTable(
+          target.toDF.queryExecution.analyzed,
+          updateColumns,
+          updateExpressions,
+          Some(c.expr))
+      case None =>
+        new UpdateTable(
+          target.toDF.queryExecution.analyzed,
+          updateColumns,
+          updateExpressions,
+          None)
+    }
+  }
+
+  protected def executeUpdate(
+      setColumns: Seq[(String, Column)],
+      condition: Option[Column]): Unit = {
```

Comment:
This is a good suggestion. since most public methods that call this will either have a String or Column, it's probably better to take it as a Column (superset of String and Column) and then convert to expression in this one location (as opposed every caller individually convert to expression). So I think we make this consistent version and change delete (in a separate PR) to be consistent.